mautic/mautic
# Support for plugin authentication

**Description**

This PR adds support for plugins to handle authentication allowing 3rd party SSO. 

There are two ways to authenticate via a plugin:

1. Via the credentials submitted through the standard login form
2. Via a 3rd party integration/SSO service such as Google, Github, Facebook

It does this by leveraging a new firewall and events.

See https://github.com/alanhartless/mautic-sso-plugin for example code until full documentation is written.

**Testing**

1. Apply the PR
2. Delete/clear Symfony's cache
3. Download and add the example plugin from https://github.com/alanhartless/mautic-sso-plugin and install into plugins (follow the readme instructions for install/setup)
4. Test logins.  Ensure standard login works, SSO login works with and without the integrations enabled, plugin authentication via the form submit with and without the integrations, and just try anything you can think of that may show signs of the changes being "insecure" :-)
